src/bthread/bthreadxx.h

```diff
@@ -0,0 +1,284 @@
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Authors: Shuo Zang (jasonszang@126.com)
+
+#ifndef BTHREAD_BTHREADXX_H
+#define BTHREAD_BTHREADXX_H
+
+#include <butil/macros.h>
+
+#ifdef BUTIL_CXX11_ENABLED
+
+#include <chrono>
+#include <functional>
+#include <memory>
+#include <system_error>
+#include <type_traits>
+#include "bthread/bthread.h"
+#include "bthread/condition_variable.h"
+#include "bthread/mtx_cv_base.h"
+
+namespace bthread {
+
+namespace detail {
+
+template<typename MatchType, typename TArg>
+using DisableOverload = std::enable_if<
+        !std::is_base_of<
+                MatchType,
+                typename std::decay<TArg>::type
+        >::value
+>;
+
+template<typename MatchType, typename TArg>
+using DisableOverloadT = typename DisableOverload<MatchType, TArg>::type;
+
+// Just for identifying bthread. There is a bthread_id_t but it is a totally different thing.
+using bthread_id = bthread_t;
```

Comment:
What does this alias work for? Naming like this may confuse users, consider removing it or using another one.